lnrpc/walletrpc/walletkit_server.go

```diff
@@ -1493,6 +1532,65 @@ func (w *WalletKit) RequiredReserve(ctx context.Context,
 	}, nil
 }
 
+// ListAddresses retrieves all the addresses along with their balance. An
+// account name filter can be provided to filter through all of the
+// wallet accounts and return the addresses of only those matching.
+func (w *WalletKit) ListAddresses(ctx context.Context,
+	req *ListAddressesRequest) (*ListAddressesResponse, error) {
+
+	addressLists, err := w.cfg.Wallet.ListAddresses(
+		req.AccountName,
+		req.ShowCustomAccounts,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a slice of accounts from addressLists map.
+	accounts := make([]*waddrmgr.AccountProperties, 0, len(addressLists))
+	for account := range addressLists {
+		accounts = append(accounts, account)
+	}
+
+	// Sort the accounts by derivation path.
+	sort.Slice(accounts, func(i, j int) bool {
+		scopeI := accounts[i].KeyScope
+		scopeJ := accounts[j].KeyScope
+		if scopeI.Purpose < scopeJ.Purpose {
```

Comment:
I think this could be slightly more readable:

```
if scopeI.Purpose == scopeJ.Purpose {
    if scopeI.Coin == scopeJ.Coin {
	acntNumI := accounts[i].AccountNumber
	acntNumJ := accounts[j].AccountNumber
	return acntNumI < acntNumJ
    }

    return scopeI.Coin < scopeJ.Coin
}

return scopeI.Purpose < scopeJ.Purpose
```